### PR TITLE
API authentication now respects application wide login_required setting

### PR DESCRIPTION
**Pull request by mtakac2**
The permission for accessing user resource is pretty simple for now: Depending on the application setting (login_required) we either allow access to everyone, or only to logged in users. 
This PR also takes care of bug, when it was not possible to access public project's work packages for anonymous user (https://www.openproject.org/work_packages/13689)

### CODE REVIEW

File: lib/api/root.rb [hunk 1]
```diff
@@ -45,7 +45,7 @@ def current_user
       end
 
       def authenticate
-        raise API::Errors::Unauthenticated.new if current_user.nil? || current_user.anonymous?
+        raise API::Errors::Unauthenticated.new if current_user.nil? || current_user.anonymous? if Setting.login_required?
```
Comment (ulferts) [inline lib/api/root.rb:48]: A test for your change would be nice.

**Comment by ulferts** [comment]
Work package got changed because we had duplicates.

File: lib/api/root.rb (hunk 1 above)
Comment (mtakac2) [inline lib/api/root.rb:48]: Yeah, I am working on redesigning tests for API endpoint. Tests for authorization / authentication will be included.

Comment (myabc) [inline lib/api/root.rb:48]: Better to include tests before we merge this.

**Pull request merged**